cpp/include/fixed_point/fixed_point.hpp

```diff
@@ -0,0 +1,489 @@
+    * @brief operator -=
+    *
+    * @tparam Rep1 Representation type of number being added to `this`
+    * @tparam Rad1 Radix (base) type of number being added to `this`
+    * @return Modified `this`
+    *---------------------------------------------------------------------------**/
+    template <typename Rep1, Radix Rad1>
+    CUDA_HOST_DEVICE_CALLABLE
+    fixed_point<Rep1, Rad1>& operator-=(fixed_point<Rep1, Rad1> const& rhs) {
+        *this = *this - rhs;
+        return *this;
+    }
+
+    /**---------------------------------------------------------------------------*
+    * @brief operator /=
+    *
+    * @tparam Rep1 Representation type of number being added to `this`
+    * @tparam Rad1 Radix (base) type of number being added to `this`
+    * @return Modified `this`
+    *---------------------------------------------------------------------------**/
+    template <typename Rep1, Radix Rad1>
+    CUDA_HOST_DEVICE_CALLABLE
+    fixed_point<Rep1, Rad1>& operator/=(fixed_point<Rep1, Rad1> const& rhs) {
+        *this = *this / rhs;
+        return *this;
+    }
+
+    /**---------------------------------------------------------------------------*
+    * @brief operator ++ (post-increment)
+    *
+    * @return Modified `this` (having added 1 to the `this`)
+    *---------------------------------------------------------------------------**/
+    CUDA_HOST_DEVICE_CALLABLE
+    fixed_point<Rep, Rad>& operator++() {
+        *this = *this + fixed_point<Rep, Rad>{1, scale_type{_scale}};
+        return *this;
+    }
+
+    /**---------------------------------------------------------------------------*
+    * @brief operator + (for adding two `fixed_point` numbers)
+    *
+    * If `_scale`s are equal, `_value`s are added
+    * If `_scale`s are not equal, number with smaller `_scale` is shifted to the
+    * greater `_scale`, and then `_value`s are added
+    *
+    * @tparam Rep1 Representation type of number being added to `this`
+    * @tparam Rad1 Radix (base) type of number being added to `this`
+    * @return resuling `fixed_point` number
+    *---------------------------------------------------------------------------**/
+    template <typename Rep1, Radix Rad1>
+    CUDA_HOST_DEVICE_CALLABLE
+    friend fixed_point<Rep1, Rad1> operator+(fixed_point<Rep1, Rad1> const& lhs,
+                                             fixed_point<Rep1, Rad1> const& rhs);
+
+    /**---------------------------------------------------------------------------*
+    * @brief operator - (for subtracting two `fixed_point` numbers)
+    *
+    * If `_scale`s are equal, `_value`s are substracted
+    * If `_scale`s are not equal, number with smaller `_scale` is shifted to the
+    * greater `_scale`, and then `_value`s are substracted
+    *
+    * @tparam Rep1 Representation type of number being added to `this`
+    * @tparam Rad1 Radix (base) type of number being added to `this`
+    * @return resuling `fixed_point` number
```

Comment:
```suggestion
    * @return The resulting `fixed_point` difference.
```

BTW, suggest installing a code spell checker for your IDE/editor.